app/pages/sections/order-items/catalogue-solutions/order-item/contextCreator.js

```diff
@@ -15,15 +15,15 @@ export const getContext = ({
   selectedPrice,
   selectedRecipients,
   errorMap,
-  solutionAlreadySelected,
+  orderItems,
 }) => {
   const errorMessages = errorMap && (errorMap.quantity || errorMap.deliveryDate)
     ? ((errorMap.quantity || {}).errorMessages || [''])
       .concat((errorMap.deliveryDate || {}).errorMessages) : undefined;
   const newbackLinkRef = selectedPrice.provisioningType === 'Patient'
     ? `${baseUrl}/organisation/${orderId}/catalogue-solutions/select/solution/price/recipients/date`
     : `${baseUrl}/organisation/${orderId}/catalogue-solutions/select/solution/price/${selectedPrice.type.toLowerCase()}/${selectedPrice.provisioningType.toLowerCase()}`;
-  const backLinkIfSolutionExist = solutionAlreadySelected
+  const backLinkIfSolutionExist = orderItems && orderItems.length > 0
     ? `${baseUrl}/organisation/${orderId}/catalogue-solutions/select/solution/`
     : `${baseUrl}/organisation/${orderId}/catalogue-solutions`;
```

Comment:
Do we not need to check that `orderItems` contains the relevant catalogue item ID? (The now badly-named `orderItemId` parameter should actually be the catalogue item ID.)